controllers/nodefeaturediscovery_finalizers.go

```diff
@@ -0,0 +1,411 @@
+	// Update CRD status to notify instance is undergoing deletion
+	_, _ = r.updateProgressingCondition(instance, "finalizers", "Foreground-Deletion")
+
+	// Attempt to delete worker DaemonSet
+	err := wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteDaemonSet(ctx, instance.ObjectMeta.Namespace, nfdWorkerApp)
+		if err != nil {
+			return false, interpretError(err, "worker DaemonSet")
+		}
+		klog.Info("Worker DaemonSet resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete master DaemonSet
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteDaemonSet(ctx, instance.ObjectMeta.Namespace, nfdMasterApp)
+		if err != nil {
+			return false, interpretError(err, "master DaemonSet")
+		}
+		klog.Info("Master DaemonSet resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the Service
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteService(ctx, instance.ObjectMeta.Namespace, nfdMasterApp)
+		if err != nil {
+			return false, interpretError(err, "nfd-master Service")
+		}
+		klog.Info("nfd-master Service resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the Role
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteRole(ctx, instance.ObjectMeta.Namespace, nfdWorkerApp)
+		if err != nil {
+			return false, interpretError(err, "nfd-worker Role")
+		}
+		klog.Info("nfd-worker Role resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the ClusterRole
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteClusterRole(ctx, instance.ObjectMeta.Namespace, nfdMasterApp)
+		if err != nil {
+			return false, interpretError(err, "nfd-master ClusterRole")
+		}
+		klog.Info("nfd-master ClusterRole resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the RoleBinding
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteRoleBinding(ctx, instance.ObjectMeta.Namespace, nfdWorkerApp)
+		if err != nil {
+			return false, interpretError(err, "nfd-worker RoleBinding")
+		}
+		klog.Info("nfd-worker RoleBinding resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the ClusterRoleBinding
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteClusterRoleBinding(ctx, instance.ObjectMeta.Namespace, nfdMasterApp)
+		if err != nil {
+			return false, interpretError(err, "ClusterRoleBinding")
+		}
+		klog.Info("ClusterRoleBinding", nfdMasterApp, "resource has been deleted.")
```

Comment:
whitespace around words?

```suggestion
		klog.Info("ClusterRoleBinding ", nfdMasterApp, " resource has been deleted.")
```